crates/observe/src/tracing_reload_handler.rs

```diff
@@ -22,7 +22,23 @@ pub(crate) fn spawn_reload_handler<T: 'static>(
         let socket_path = format!("/tmp/log_filter_override_{name}_{id}.sock");
         tracing::warn!(file = socket_path, "open log filter reload socket");
         let handle = SocketHandle {
-            listener: UnixListener::bind(&socket_path).expect("socket handle is unique"),
+            listener: match UnixListener::bind(&socket_path) {
```

Comment:
What's the reason to prefer this more complex approach over just always trying to delete the file before opening the socket? That would have been a one liner with effectively the same outcome.

Reply:
I think I did some misunderstand of our discussion on Slack, now reading it again it looks that one line change solution is enough. I have no special reasoning for this bind retry approach beyond just logging the errors.